packages/web-components/src/switch/switch.styles.ts

```diff
@@ -224,21 +224,26 @@ export const switchStyles: (context: ElementDefinitionContext, definition: Switc
         :host(.checked:not(.disabled)) .switch slot[name="switch"] {
           fill: ${SystemColors.HighlightText};
         }
-        :host(.checked:not(.disabled):hover) .switch slot[name='switch'] {
+        :host(.checked:not(.disabled):hover) .switch slot[name='switch'],
+        :host(.checked:not(.disabled):${focusVisible}) .switch slot[name='switch'] {
           fill: ${SystemColors.Highlight};
         }
         :host(:${focusVisible}) .switch {
           forced-color-adjust: none;
-          background: ${SystemColors.Field}; 
+          background: ${SystemColors.Field};
           border-color: ${SystemColors.Highlight};
           outline-color: ${SystemColors.FieldText};
         }
         :host(.disabled) {
           opacity: 1;
         }
-        :host(.disabled) slot[name='switch'] {
+        :host(.disabled) slot[name='switch'],
+        :host(.disabled) .label,
+        :host(.disabled) ::slotted([slot="checked-message"]),
+        :host(.disabled) ::slotted([slot="unchecked-message"]) {
           forced-color-adjust: none;
-          fill: ${SystemColors.GrayText};
+          color: ${SystemColors.GrayText};
+          fill: currentcolor;
```

Comment:
Check if this is needed. Should come from regular styles.

Reply:
This is still needed